tests/test_unittest/test_gtscript_frontend.py

```diff
@@ -477,6 +480,118 @@ def definition_func(field: gtscript.Field[float]):
         )
 
 
+class TestRegions:
+    def test_one_interval_only(self):
+        def stencil(in_f: gtscript.Field[np.float_]):
+            with computation(PARALLEL), interval(...), horizontal(region[I[0:3], :]):
+                in_f = 1.0
+
+        def_ir = parse_definition(
+            stencil, name=inspect.stack()[0][3], module=self.__class__.__name__
+        )
+
+        assert len(def_ir.computations) == 1
+        assert isinstance(def_ir.computations[0].body.stmts[0], gt_ir.HorizontalIf)
+
+    def test_one_interval_only_single(self):
+        def stencil(in_f: gtscript.Field[np.float_]):
+            with computation(PARALLEL), interval(...), horizontal(region[I[0], :]):
+                in_f = 1.0
+
+        def_ir = parse_definition(
+            stencil, name=inspect.stack()[0][3], module=self.__class__.__name__
+        )
+
+        assert len(def_ir.computations) == 1
+        assert def_ir.computations[0].body.stmts[0].intervals["I"].is_single_index
+
+    def test_from_external(self):
+        def stencil(in_f: gtscript.Field[np.float_]):
+            from gt4py.__externals__ import i1
+
+            with computation(PARALLEL), interval(...), horizontal(region[i1, :]):
+                in_f = 1.0
+
+        def_ir = parse_definition(
+            stencil,
+            name=inspect.stack()[0][3],
+            module=self.__class__.__name__,
+            externals={"i1": I[1]},
+        )
+
+        assert len(def_ir.computations) == 1
+        assert (
+            def_ir.computations[0].body.stmts[0].intervals["I"].start.level
+            == gt_ir.LevelMarker.START
+        )
+        assert def_ir.computations[0].body.stmts[0].intervals["I"].start.offset == 1
+        assert def_ir.computations[0].body.stmts[0].intervals["I"].is_single_index
+
+    def test_multiple_inline(self):
+        def stencil(in_f: gtscript.Field[np.float_]):
+            with computation(PARALLEL), interval(...):
+                in_f = in_f + 1.0
+                with horizontal(region[I[0], :], region[:, J[-1]]):
+                    in_f = 1.0
+
+        def_ir = parse_definition(
+            stencil, name=inspect.stack()[0][3], module=self.__class__.__name__
+        )
+
+        assert len(def_ir.computations[0].body.stmts) == 3
+
+    def test_inside_function(self):
+        @gtscript.function
+        def region_func():
+            from gt4py.__externals__ import ie
+
+            field = 0.0
+            with horizontal(region[ie, :]):
+                field = 1.0
+
+            return field
+
+        def stencil(in_f: gtscript.Field[np.float_]):
+            with computation(PARALLEL), interval(...):
+                in_f = region_func()
+
+        def_ir = parse_definition(
+            stencil,
+            name=inspect.stack()[0][3],
+            module=self.__class__.__name__,
+            externals={"ie": I[-1]},
+        )
+
+        # TODO: Does this need to assert anything?
```

Comment:
It doesn't need to assert anything. Without any assertions it will fail in case of any uncaught exceptions.

Reply:
👍